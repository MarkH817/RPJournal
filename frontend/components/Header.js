import React from 'react'
import Link from 'next/link'

import { useUser } from './hooks/useUser'
import StyledHeader from './styles/Header'
import Logout from './Logout'
import Nav from './Nav'

export function Header () {
  const { loading, error, data } = useUser()

  return (
    <StyledHeader>
      <div className='bar'>
        <h1>
          <Link href='/'>
            <a>RPJournal</a>
          </Link>
        </h1>
      </div>

      <div className='sub-bar'>
        <Nav>
          <ul>
            <li>
              <Link href='/'>
                <a>Home</a>
              </Link>
            </li>
            <li>
              {loading || error || !data.me ? (
                <Link href='/login'>
                  <a>Login</a>
                </Link>
              ) : (
                <Logout />
              )}
            </li>
          </ul>
        </Nav>
      </div>
    </StyledHeader>
  )
}

export default Header
