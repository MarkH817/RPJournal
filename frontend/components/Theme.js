import React from 'react'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import 'modern-normalize'
import 'nprogress/nprogress.css'
import '@reach/skip-nav/styles.css'
import 'draft-js/dist/Draft.css'
import 'draftail/dist/draftail.css'
import 'typeface-open-sans'
import 'typeface-sarabun'

const theme = {
  white: 'rgb(255,255,255)',
  offWhite: 'rgb(225,225,225)',
  lightGray: 'rgb(175,175,175)',
  gray: 'rgb(125,125,125)',
  darkGray: 'rgb(75,75,75)',
  black: 'rgb(25,25,25)',
  primaryColor: '#ea45ba',
  accentColor: '#1d5e81',
  headerFont: `'Sarabun', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`,
  baseFont: `'Open Sans', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`
}

const GlobalStyles = createGlobalStyle`
  * {
    outline-width: 2px;
  }

  body {
    color: ${props => props.theme.black};
    font-family: ${props => props.theme.baseFont};
    font-size: 1em;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }

  .detail-notes {
    line-height: 1.5;
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
