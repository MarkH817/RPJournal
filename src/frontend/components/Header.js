import React from 'react'
import styled from 'styled-components'

import Nav from './Nav'

const StyledHeader = styled.header`
  .bar {
    align-items: stretch;
    border-bottom: 0.625em solid ${props => props.theme.black};
    display: grid;
    grid-template-columns: auto 1fr;
    justify-content: space-between;

    h1 {
      font-family: ${props => props.theme.baseFont};
      font-size: 4em;
      margin-left: 0.5em;
      position: relative;

      @media (max-width: 1300px) {
        margin: 0;
        text-align: center;
      }

      a {
        padding: 0.125em 0.25em;
        text-decoration: none;
      }
    }

    @media (max-width: 1300px) {
      grid-template-columns: 1fr;
      justify-content: center;
    }
  }

  .sub-bar {
    display: grid;
    grid-template-columns: 1fr auto;
    border-bottom: 1px solid ${props => props.theme.black};
  }
`

const Header = props => (
  <StyledHeader>
    <div className='bar'>
      <h1>
        <a href='/'>RPJournal</a>
      </h1>
    </div>

    <div className='sub-bar'>
      <Nav />
    </div>
  </StyledHeader>
)

export default Header
