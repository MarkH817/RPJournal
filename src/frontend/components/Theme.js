import React from 'react'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import 'modern-normalize'
import 'nprogress/nprogress.css'
import '@reach/skip-nav/styles.css'
import 'draft-js/dist/Draft.css'
import 'draftail/dist/draftail.css'
import 'typeface-open-sans'
import 'typeface-sarabun'

const theme = {
  white: '#ffffff',
  offWhite: '#dfdfdf',
  lightGray: '#c3c3c3',
  gray: '#767676',
  darkGray: '#525252',
  black: '#353535',
  primaryColor: '#ea45ba',
  accentColor: '',
  headerFont: `'Sarabun', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`,
  baseFont: `'Open Sans', 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif`
}

const GlobalStyles = createGlobalStyle`
  body {
    color: ${props => props.theme.black};
    font-family: ${props => props.theme.baseFont};
    font-size: 1em;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
