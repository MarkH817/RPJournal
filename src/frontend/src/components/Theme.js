import React from 'react'
import '@reach/skip-nav/styles.css'
import { createGlobalStyle, ThemeProvider } from 'styled-components'

const theme = {
  black: '#353535',
  primaryColor: 'royalblue'
}

const GlobalStyles = createGlobalStyle`
  *,
  *:after,
  *:before {
    box-sizing: border-box;
  }

  body {
    color: ${props => props.theme.black};
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: 1em;
    margin: 0;
    padding: 0;
  }

  a {
    color: ${props => props.theme.primaryColor};
  }
`

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    <React.Fragment>
      <GlobalStyles />

      {children}
    </React.Fragment>
  </ThemeProvider>
)

export default Theme
