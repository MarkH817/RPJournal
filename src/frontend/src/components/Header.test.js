import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Header from './Header'
import Theme from './Theme'

describe('Header', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Theme>
          <Header />
        </Theme>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
