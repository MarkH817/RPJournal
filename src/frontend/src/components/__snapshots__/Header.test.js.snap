// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c1 {
  font-size: 1.5em;
}

.c1 a,
.c1 button {
  background-color: transparent;
  border: none;
  display: block;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  padding: 0.25em 0.5em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 .menu-button {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  width: 100%;
}

.c1 .menu {
  -webkit-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
  max-height: 0;
  overflow: hidden;
}

.c1 .menu.active {
  max-height: 25em;
}

.c1 .menu ul {
  border-top: 1px solid #353535;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c1 .menu a,
.c1 .menu button {
  border-bottom: 1px solid #353535;
}

.c0 .bar {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-bottom: 0.625em solid #353535;
  display: grid;
  grid-template-columns: auto 1fr;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 .bar h1 {
  font-size: 4em;
  margin-left: 0.5em;
  position: relative;
}

.c0 .bar h1 a {
  padding: 0.125em 0.25em;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 .sub-bar {
  display: grid;
  grid-template-columns: 1fr auto;
  border-bottom: 1px solid #353535;
}

@media screen and (min-width:55em) {
  .c1 .menu-button {
    display: none;
  }

  .c1 .menu {
    max-height: none;
  }

  .c1 .menu ul {
    border: 0;
    margin: 0 0 0 -0.25em;
  }

  .c1 .menu li {
    display: inline-block;
    margin: 0 0.25em;
  }

  .c1 .menu a {
    border: 0;
  }
}

@media (max-width:1300px) {
  .c0 .bar h1 {
    margin: 0;
    text-align: center;
  }
}

@media (max-width:1300px) {
  .c0 .bar {
    grid-template-columns: 1fr;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<header
  className="c0"
>
  <div
    className="bar"
  >
    <h1>
      <a
        href="/"
      >
        RPJournal
      </a>
    </h1>
  </div>
  <div
    className="sub-bar"
  >
    <nav
      className="c1"
    >
      <button
        className="menu-button"
        onClick={[Function]}
      >
        <span>
          Menu
        </span>
      </button>
      <div
        className="menu"
        onClick={[Function]}
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/journal"
            >
              Journal
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</header>
`;
